Preserve metadata on S3 when setting Cache-Control

glogiotatidis opened the pull request:
Had to modify the metadata set command to ensure that `content-type` is preserved when we set cache-control headers. Since i was doing that i also preserved `mtime`.

This is a bit more complex than it should be, just because `aws s3api` command cannot do it by itself.

glogiotatidis commented:
Will merge this to fix staging install, but would love to hear comments @pmac @duallain

pmac commented:
Is the reason we have to do the `copy-object` call that you can't update metadata like cache-control on an existing object? In bedrock I just do 2 syncs, one for the immutable files and one for everything else. Perhaps that would be more simple here as well, or maybe something more is going on?

glogiotatidis commented:
> Is the reason we have to do the `copy-object` call that you can't update metadata like cache-control on an existing object? In bedrock I just do 2 syncs, one for the immutable files and one for everything else. Perhaps that would be more simple here as well, or maybe something more is going on?

That's the idea here too. We used to do two syncs exactly as you describe, but the `aws sync` was *too* slow. We changed to using `rclone` which is way faster but does not support setting of the cache-control header, so we do this `copy-object` trick. 

I simplified and made more robust things here https://github.com/mozmeao/lumbergh/commit/a2e0b6e04a1ce6f7b11e95a13cc9a6d929cf101a
